### DISCUSSION TITLE
绘制PolygonGraphics面outlineWidth属性不生效

### DISCUSSION THREAD

**Discussion by jiege15121313** [zouyaoji/vue-cesium#308] [General]
![Snipaste](https://user-images.githubusercontent.com/115357712/199903090-b5cca1c8-702d-43e4-b2b1-ea3c33e32a16.png)

**Comment by jiege15121313** [comment]
![166145](https://user-images.githubusercontent.com/115357712/199909740-be37bf92-9b4b-42f0-b8b2-9cff0291673d.png)
还有同样经纬度绘制出来的图，比wmts绘制出来的小一圈是怎么回事呢？

**Comment by zouyaoji** [comment]
1、outline这是webgl在windows上的bug，无解哈。
2、可能是你的图不一样。

**Reply by jiege15121313** [reply]
好的，还有一个问题想请教一下你，VcDrawings的preRenderDatas不支持异步的话，在viwer没有重新加载的情况下有没有方法能让polygonDrawing根据接口传的经纬度来动态画面呢

**Reply by zouyaoji** [reply]
看看这个可以满足你需求不。
https://github.com/zouyaoji/vue-cesium/discussions/188#discussioncomment-2543262

后续还想改变 preRenderDatas ，可以通过模板引用 ref ，调用组件的reload方法试试看看

**Reply by jiege15121313** [reply]
经纬度信息是通过填写表单之后传输过来，所以v-if这个方法可能难以实现，我也尝试过调用组件的reload方法，但是视图并没有更新，所以我猜想会不会是需要整个viewer重载才会加载preRenderDatas ？后面我尝试使用VcGraphicsPolygon，但是会出现比地图小一圈的问题（使用preRenderDatas 没有发现这个问题）

**Reply by zouyaoji** [reply]
这么说来，你说用那个VcGraphicsPolygon加出来不同可能是arcType不同。
vc-drawings的面是用 VcPolygon加载的，你可以试试这个组件。

reload的情况我回头也测测看看吧。

**Reply by jiege15121313** [reply]
好的，感谢

**Reply by jiege15121313** [reply]
VcPolygon这个组件在文档上怎么没找到呢

**Reply by zouyaoji** [reply]
https://zouyaoji.top/vue-cesium/#/zh-CN/component/primitives/vc-collection-primitive#vcpolygon

**Reply by jiege15121313** [reply]
VcPolygon是否能设置outline呢

**Reply by zouyaoji** [reply]
目前还没支持outline，下个版本可以考虑加一下。
你着急可以自己去实现然后提个pr。

https://github.com/zouyaoji/vue-cesium/blob/dev/packages/shared/src/PolygonPrimitive.ts#L159

**Reply by jiege15121313** [reply]
我看了一下，看不太懂您的源码哈哈

**Reply by luoyuchen0417** [reply]
> 目前还没支持outline，下个版本可以考虑加一下。 你着急可以自己去实现然后提个pr。
> 
> https://github.com/zouyaoji/vue-cesium/blob/dev/packages/shared/src/PolygonPrimitive.ts#L159

请问现在支持了吗，我最近试了下宽度设置还是没有成功
